src/scanner-engine.ts

```diff
@@ -54,30 +50,33 @@ export async function fetchScannerEngine(properties: ScannerProperties) {
 
   properties[ScannerProperty.SonarScannerWasEngineCacheHit] = 'false';
 
-  const { archivePath, unarchivePath: scannerEnginePath } = await getCacheDirectories(properties, {
+  const { archivePath } = await getCacheDirectories(properties, {
     checksum,
     filename,
   });
   const url = downloadUrl ?? API_V2_SCANNER_ENGINE_ENDPOINT;
   log(LogLevel.DEBUG, `Starting download of Scanner Engine`);
   await download(url, archivePath);
-  log(LogLevel.INFO, `Downloaded Scanner Engine to ${scannerEnginePath}`);
+  log(LogLevel.INFO, `Downloaded Scanner Engine to ${archivePath}`);
 
-  await validateChecksum(archivePath, checksum);
+  try {
+    await validateChecksum(archivePath, checksum);
+  } catch (error) {
+    await fsExtra.remove(archivePath);
```

Comment:
We weren't removing the file when the checksum didn't match, so a second attempt at running the scanner would actually use the file (since it wasn't removed, the bootstrapper things it's cached)

Reply:
good catch! but we are in the same situation if the removal fails. I would `validateChecksum` also in the `getCacheFileLocation` method. Right now only check for file existence.

Reply:
sorry I merged before I saw this. I'll create a dogfooding PR and address this